### ISSUE TITLE
[camera] Cannot pass maxDuration from startVideoRecording to CameraPlatform.instance.startVideoCapturing

### ISSUE THREAD

**Issue by Ne0n11** [flutter/flutter#150925]
### What package does this bug report belong to?

camera

### What target platforms are you seeing this bug on?

Android, iOS

### Have you already upgraded your packages?

Yes

### Dependency versions

<details><summary>pubspec.lock</summary>

```lock
camera:
    dependency: "direct main"
    description:
      name: camera
      sha256: "2170a943dcb67be2af2c6bcda8775e74b41d4c02d6a4eb10bdc832ee185c4eea"
      url: "https://pub.dev"
    source: hosted
    version: "0.11.0+1"
  camera_android_camerax:
    dependency: transitive
    description:
      name: camera_android_camerax
      sha256: "59967e6d80df9d682a33b86f228cc524e6b52d6184b84f6ac62151dd98bd1ea0"
      url: "https://pub.dev"
    source: hosted
    version: "0.6.5+2"
  camera_avfoundation:
    dependency: transitive
    description:
      name: camera_avfoundation
      sha256: "7d021e8cd30d9b71b8b92b4ad669e80af432d722d18d6aac338572754a786c15"
      url: "https://pub.dev"
    source: hosted
    version: "0.9.16"
  camera_platform_interface:
    dependency: transitive
    description:
      name: camera_platform_interface
      sha256: a250314a48ea337b35909a4c9d5416a208d736dcb01d0b02c6af122be66660b0
      url: "https://pub.dev"
    source: hosted
    version: "2.7.4"
  camera_web:
    dependency: transitive
    description:
      name: camera_web
      sha256: "9e9aba2fbab77ce2472924196ff8ac4dd8f9126c4f9a3096171cd1d870d6b26c"
      url: "https://pub.dev"
    source: hosted
    version: "0.3.3"
```

</details>

### Steps to reproduce

1. Use startVideoRecording method
2. See there is no option to pass maxDuration for video recording

### Expected results

startVideoRecording method includes optional maxDuration parameter, which then is passed to CameraPlatform.instance.startVideoCapturing(VideoOptions(maxDuration))

### Actual results

1. Use startVideoRecording method
2. See there is no option to pass maxDuration for video recording

### Code sample

<details open><summary>Code sample</summary>

```dart```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.19.6, on macOS 14.5 23F79 darwin-x64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.1)
[✓] VS Code (version 1.90.2)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by stuartmorgan** [comment]
This would be best implemented at the client level (or in a third-party utility package), by starting your own timer and ending the recording at that point. The only reason `image_picker` provides a `maxDuration` is that there is no ability for the caller to directly control the recording.

Closing as out of scope for the plugin.

**Comment by Ne0n11** [comment]
> This would be best implemented at the client level (or in a third-party utility package), by starting your own timer and ending the recording at that point. The only reason `image_picker` provides a `maxDuration` is that there is no ability for the caller to directly control the recording.
> 
> Closing as out of scope for the plugin.

The issue is about camera package, not image_picker. 

This is code now 
```dart
  Future<void> startVideoRecording(
      {onLatestImageAvailable? onAvailable}) async {
    _throwIfNotInitialized('startVideoRecording');
    if (value.isRecordingVideo) {
      throw CameraException(
        'A video recording is already started.',
        'startVideoRecording was called when a recording is already started.',
      );
    }

    void Function(CameraImageData image)? streamCallback;
    if (onAvailable != null) {
      streamCallback = (CameraImageData imageData) {
        onAvailable(CameraImage.fromPlatformInterface(imageData));
      };
    }

    try {
      await CameraPlatform.instance.startVideoCapturing(
          VideoCaptureOptions(_cameraId, streamCallback: streamCallback));
      value = value.copyWith(
          isRecordingVideo: true,
          isRecordingPaused: false,
          recordingOrientation: Optional<DeviceOrientation>.of(
              value.lockedCaptureOrientation ?? value.deviceOrientation),
          isStreamingImages: onAvailable != null);
    } on PlatformException catch (e) {
      throw CameraException(e.code, e.message);
    }
  }
```

and this is what I propose
```dart
  Future<void> startVideoRecording({
    onLatestImageAvailable? onAvailable,
    Duration? maxDuration,
  }) async {
    _throwIfNotInitialized('startVideoRecording');
    if (value.isRecordingVideo) {
      throw CameraException(
        'A video recording is already started.',
        'startVideoRecording was called when a recording is already started.',
      );
    }

    void Function(CameraImageData image)? streamCallback;
    if (onAvailable != null) {
      streamCallback = (CameraImageData imageData) {
        onAvailable(CameraImage.fromPlatformInterface(imageData));
      };
    }

    try {
      await CameraPlatform.instance.startVideoCapturing(
        VideoCaptureOptions(
          _cameraId,
          streamCallback: streamCallback,
          maxDuration: maxDuration,
        ),
      );

      value = value.copyWith(
          isRecordingVideo: true,
          isRecordingPaused: false,
          recordingOrientation: Optional<DeviceOrientation>.of(
              value.lockedCaptureOrientation ?? value.deviceOrientation),
          isStreamingImages: onAvailable != null);
    } on PlatformException catch (e) {
      throw CameraException(e.code, e.message);
    }
  }
```

**Comment by Ne0n11** [comment]
So, its already implemented in Platform, but its just not passed from camera function itself. @stuartmorgan

**Comment by stuartmorgan** [comment]
> The issue is about camera package, not image_picker.

I understand that; I was contrasting the `camera` case with another plugin where we do implement this, and explaining why the decision is different here.

> So, its already implemented in Platform, but its just not passed from camera function itself.

It was added to the platform interface, but never implemented on any platform. It shouldn't have been added in the first place; I've filed https://github.com/flutter/flutter/issues/150959 to partially clean that up.
